File: languages/cpp/7sorting/21sortanarraywithtwotypesofelements.cpp

#include <bits/stdc++.h>
using namespace std;

void sort(int arr[],int n){
    int temp[n],i=0;
    
    for(int j=0;j<n;j++)
        if(arr[j]<0)
            temp[i++]=arr[j];
    for(int j=0;j<n;j++)
        if(arr[j]>=0)
            temp[i++]=arr[j];
    for(int j=0;j<n;j++)
        arr[j]=temp[j];
}

int main() {
	
    int arr[]={13,-12,18,-10};
	
	int n=sizeof(arr)/sizeof(arr[0]);
	
	sort(arr,n);
	
	for(int x:arr)
	    cout<<x<<" ";
}

File: languages/cpp/7sorting/21sortanarraywithtwotypesofelements.java

import java.util.*;
import java.lang.*;
import java.io.*;

class Solution
{
    public static void main (String[] args) 
    {
        int arr[] = new int[]{13,-12,18,-10};
        
        int n = arr.length;
        
        sort(arr,n);
	
	    for(int x:arr)
            System.out.print(x+" ");
        
    }
    
    static void sort(int arr[],int n){
        int[] temp=new int[n];int i=0;
        
        for(int j=0;j<n;j++)
            if(arr[j]<0)
                temp[i++]=arr[j];
        for(int j=0;j<n;j++)
            if(arr[j]>=0)
                temp[i++]=arr[j];
        for(int j=0;j<n;j++)
            arr[j]=temp[j];
    }
}
